AttributeError: 'ArchiveEntry' object has no attribute '_pointer' when getting the entry format_name

Reproducer:

```python3
import io
import os
import tarfile
import libarchive

file_name = "tar-with-single-file.tar.bz2"
if os.path.exists(file_name):
    raise RuntimeError("Will not overwrite existing file!")

with tarfile.open(file_name, "w:bz2") as archive:
    archive.addfile(tarfile.TarInfo("foo"), fileobj=io.BytesIO(b"bar"))

try:
    with libarchive.file_reader(file_name) as archive:
        entry = next(iter(archive))
        print("format name:", entry.format_name)
    print("Done")
finally:
    os.remove(file_name)
```

Error:

```
Traceback (most recent call last):
  File "test_format_name.py", line 16, in <module>
    print("format name:", entry.format_name)
  File "~/.local/lib/python3.10/site-packages/libarchive/entry.py", line 438, in format_name
    return ffi.format_name(self._pointer)
AttributeError: 'ArchiveEntry' object has no attribute '_pointer'
```

It might be a good idea to add static analysis such as pylint, which would have noticed this kind of error. Unfortunately, there are a lot of false positives because of the ffi module, which I had to filter:

```
> pylint libarchive/*py | grep ': E' | grep -v "[Mm]odule 'libarchive.ffi"
libarchive/entry.py:374:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
libarchive/entry.py:438:31: E1101: Instance of 'ArchiveEntry' has no '_pointer' member (no-member)
libarchive/ffi.py:70:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
```

--- a/libarchive/entry.py
+++ b/libarchive/entry.py
@@ -433,10 +433,6 @@ class ArchiveEntry:
     def rdevminor(self, value):
         ffi.entry_set_rdevminor(self._entry_p, value)
 
-    @property
-    def format_name(self):
-        return ffi.format_name(self._archive_p)
-
 
 class ConsumedArchiveEntry(ArchiveEntry):
 

--- a/libarchive/entry.py
+++ b/libarchive/entry.py
@@ -435,7 +435,7 @@ class ArchiveEntry:
 
     @property
     def format_name(self):
-        return ffi.format_name(self._pointer)
+        return ffi.format_name(self._archive_p)
 
 
 class ConsumedArchiveEntry(ArchiveEntry):

--- a/libarchive/read.py
+++ b/libarchive/read.py
@@ -39,6 +39,10 @@ class ArchiveRead:
         count = ffi.filter_count(self._pointer)
         return [ffi.filter_name(self._pointer, i) for i in range(count - 1)]
 
+    @property
+    def format_name(self):
+        return ffi.format_name(self._pointer)
+
 
 @contextmanager
 def new_archive_read(format_name='all', filter_name='all', passphrase=None):

--- a/tests/test_rwx.py
+++ b/tests/test_rwx.py
@@ -26,6 +26,8 @@ def test_buffers(tmpdir):
     # Read the archive and check that the data is correct
     with libarchive.memory_reader(buf) as archive:
         check_archive(archive, tree)
+        assert archive.format_name == b'GNU tar format'
+        assert archive.filter_names == [b'xz']
 
     # Extract the archive in tmpdir and check that the data is intact
     with in_dir(tmpdir.strpath):
@@ -50,6 +52,8 @@ def test_fd(tmpdir):
     archive_file.seek(0)
     with libarchive.fd_reader(fd) as archive:
         check_archive(archive, tree)
+        assert archive.format_name == b'GNU tar format'
+        assert archive.filter_names == [b'bzip2']
 
     # Extract the archive in tmpdir and check that the data is intact
     archive_file.seek(0)
@@ -73,6 +77,8 @@ def test_files(tmpdir):
     # Read the archive and check that the data is correct
     with libarchive.file_reader(archive_path) as archive:
         check_archive(archive, tree)
+        assert archive.format_name == b'POSIX ustar format'
+        assert archive.filter_names == [b'gzip']
 
     # Extract the archive in tmpdir and check that the data is intact
     with in_dir(tmpdir.strpath):
@@ -95,6 +101,8 @@ def test_custom_writer_and_stream_reader():
     # Read the archive and check that the data is correct
     with libarchive.stream_reader(stream, 'zip') as archive:
         check_archive(archive, tree)
+        assert archive.format_name == b'ZIP 2.0 (deflation)'
+        assert archive.filter_names == []
 
 
 @patch('libarchive.ffi.write_fail')

